### PR TITLE
[GOBBLIN-1862] Allow reference count to be 0 but no less

### PR DESCRIPTION
**Pull request by AndyJiang99**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-1862] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1862

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
Due to possible issues upstream reporting incorrect reference counts for number of records, there could be times when the reference count reported to be 0.
In the current case, if reference tier counter is reported to be 0 but there are counts in other PCM tiers, an exception is thrown, however is silently swallowed, thus, causing the watermark to not move forward in this case.
This PR fixes said issue, allowing reference count to be 0 but no less. For completeness percentage calculations in such cases, the completeness will be reported as positive infinity as any number divided by double 0 is positive infinity.

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Unit test added to check that no exceptions are encountered when reference count is 0

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-completeness/src/main/java/org/apache/gobblin/completeness/verifier/KafkaAuditCountVerifier.java [hunk 1]
```diff
@@ -217,7 +217,7 @@ private static void validateTierCounts(String datasetName, long beginInMillis, l
         throw new IOException(String.format("Reference tier %s audit count cannot be retrieved for dataset %s between %s and %s", refTier, datasetName, beginInMillis, endInMillis));
       }
       long refCount = countsByTier.get(refTier);
-      if(refCount <= 0) {
+      if (refCount < 0) {
         throw new IOException(String.format("Reference tier %s count cannot be less than or equal to zero", refTier));
```
Comment (vikrambohra) [inline gobblin-completeness/src/main/java/org/apache/gobblin/completeness/verifier/KafkaAuditCountVerifier.java:221]: Change exception log line to match the if condition "less than zero"

File: gobblin-completeness/src/main/java/org/apache/gobblin/completeness/verifier/KafkaAuditCountVerifier.java [hunk 2]
```diff
@@ -217,7 +217,7 @@ private static void validateTierCounts(String datasetName, long beginInMillis, l
         throw new IOException(String.format("Reference tier %s audit count cannot be retrieved for dataset %s between %s and %s", refTier, datasetName, beginInMillis, endInMillis));
       }
       long refCount = countsByTier.get(refTier);
-      if(refCount <= 0) {
+      if (refCount < 0) {
```
Comment (vikrambohra) [inline gobblin-completeness/src/main/java/org/apache/gobblin/completeness/verifier/KafkaAuditCountVerifier.java:220]: I think we should add a log.warn when refCount == 0. wdyt?

File: gobblin-completeness/src/test/java/org/apache/gobblin/completeness/verifier/KafkaAuditCountVerifierTest.java [hunk 3]
```diff
@@ -113,6 +113,20 @@ public void testTotalCountCompleteness() throws IOException {
...
+    // Check validation tiers for exceptions
+    client.setTierCounts(ImmutableMap.of(
+        SOURCE_TIER, 990L,
+        REFERENCE_TIERS, 0L,
+        TOTAL_COUNT_REF_TIER_0, 0L,
+        TOTAL_COUNT_REF_TIER_1, 0L
+    ));
+    try {
+      verifier.calculateCompleteness(topic, 0L, 0L)
+          .get(KafkaAuditCountVerifier.CompletenessType.TotalCountCompleteness);
```
Comment (vikrambohra) [inline gobblin-completeness/src/test/java/org/apache/gobblin/completeness/verifier/KafkaAuditCountVerifierTest.java:126]: Your test case only covers TotalCountCompleteness. Add a new test for ClassicCompleteness

**Review by vikrambohra — CHANGES_REQUESTED** [review]

**Comment by vikrambohra** [comment]
>   In the current case, if reference tier counter is reported to be 0, an exception is thrown, however is silently swallowed. Thus, causing the watermark to not move forward in this case or in the case of a quiet topic for hours where there are no records.

"or in the case of a quiet topic for hours where there are no records." this statement is not entirely correct. When there are zero counts from all tiers the PCM api returns an empty map and we proceed with the CW update. this was fixed in https://github.com/apache/gobblin/pull/3698
Please fix description

**Review by vikrambohra — APPROVED** [review]
lgtm

**Review by ZihanLi58 — COMMENTED** [review]
+1

**Review by ZihanLi58 — APPROVED** [review]

**Pull request merged**